### PR TITLE
Initial implementation of joda-to-java-time-bridge 

### PR DESCRIPTION
**Pull request by haozhun**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>joda-to-java-time-bridge</artifactId>
+    <version>0.1-SNAPSHOT</version>
```
Comment (dain) [inline pom.xml]: `1-SNAPSHOT`

File: README.md [hunk 2]
```diff
@@ -1 +1,15 @@
-# joda-to-java-time-bridge
\ No newline at end of file
+# joda-to-java-time-bridge
+
+[![Build Status](https://travis-ci.org/airlift/joda-to-java-time-bridge.svg?branch=master)](https://travis-ci.org/airlift/joda-to-java-time-bridge)
+
+Mixed use of joda and java.time results in two different versions of
+tzdata being used at the same time. This inconsistencies can often
```
Comment (dain) [inline README.md]: I'd use `time zone database` instead of `tzdata`

Comment (dain) [reply README.md]: I would remove the word `often`

File: README.md [hunk 3]
```diff
@@ -1 +1,15 @@
...
+lead to result that is incorrect in any tzdata version. For example,
```
Comment (dain) [inline README.md]: s/in/with

File: README.md [hunk 4]
```diff
@@ -1 +1,15 @@
...
+operations as simple as casting a string representation of zoned
+date time to unixtime and back will not roundtrip.
+
+joda-to-java-time-bridge makes it possible to mix usage of joda and
```
Comment (dain) [inline README.md]: capitalize Joda

File: README.md [hunk 5]
```diff
@@ -1 +1,15 @@
...
+java.time without the above problem. Implementation wise, it makes joda
```
Comment (dain) [inline README.md]: maybe:
```
`java.time`, without the above problem, by changing Joda to delegate to the time zone database in `java.time`.
```

File: README.md [hunk 6]
```diff
@@ -1 +1,15 @@
-# joda-to-java-time-bridge
\ No newline at end of file
+# joda-to-java-time-bridge
+
+[![Build Status](https://travis-ci.org/airlift/joda-to-java-time-bridge.svg?branch=master)](https://travis-ci.org/airlift/joda-to-java-time-bridge)
+
+Mixed use of joda and java.time results in two different versions of
```
Comment (dain) [inline README.md]: Should we put backticks around `java.time`?

Comment (electrum) [reply README.md]: Yes, and use the correct name Joda-Time

File: README.md (hunk 2 above)
Comment (electrum) [reply README.md]: > results in two different zone databases being used at the same time

File: README.md (hunk 6 above)
Comment (electrum) [inline README.md]:
```
Mixed use of Joda-Time and `java.time` results in two different time zone
databases being used at the same time. This inconsistency can lead to
incorrect results. For example, operations as simple as casting a zoned
date time to a Unix time and back will not round-trip.

File: README.md (hunk 4 above)
Comment (electrum) [reply README.md]:
```
joda-to-java-time-bridge makes it possible to mix usages of Joda-Time and
`java.time` without the above problem, by changing Joda-Time to delegate
to the time zone database in `java.time` (the one built into the JVM).

File: README.md [hunk 7]
```diff
@@ -1 +1,15 @@
...
+tzdata being used at the same time. This inconsistencies can often
+lead to result that is incorrect in any tzdata version. For example,
+operations as simple as casting a string representation of zoned
+date time to unixtime and back will not roundtrip.
+
+joda-to-java-time-bridge makes it possible to mix usage of joda and
+java.time without the above problem. Implementation wise, it makes joda
+delegate to java.time for tz rules.
+
+It is expected to work with any reasonable version of joda.
```
Comment (electrum) [inline README.md]:
```
It is expected to work with any reasonable version of Joda-Time.

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    <packaging>jar</packaging>
+
+    <name>joda-to-java-time-bridge</name>
+    <description>Make joda delegate to java.time for time zone rules</description>
```
Comment (electrum) [inline pom.xml]: Joda-Time

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    <url>https://github.com/airlift/joda-to-java-time-bridge</url>
+
+    <parent>
+        <groupId>io.airlift</groupId>
+        <artifactId>airbase</artifactId>
+        <version>82</version>
```
Comment (electrum) [inline pom.xml]: Use 88

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    </parent>
+
+    <inceptionYear>2018</inceptionYear>
+
+    <properties>
+        <air.check.skip-extended>true</air.check.skip-extended>
```
Comment (electrum) [inline pom.xml]: We should disable specific checkers rather than all of them

File: pom.xml [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    </properties>
+
+    <organization>
```
Comment (electrum) [inline pom.xml]: This shouldn't be needed as it inherits from the parent POM

File: pom.xml [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+        <name>Airlift</name>
+    </organization>
+
+    <licenses>
```
Comment (electrum) [inline pom.xml]: Same

File: src/main/java/io/airlift/jodabridge/JdkBasedDateTimeZone.java [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+import java.time.zone.ZoneOffsetTransition;
+import java.time.zone.ZoneRules;
+import java.util.Objects;
+
+public class JdkBasedDateTimeZone
+        extends DateTimeZone
+{
+    private final ZoneRules zoneRules;
+
+    JdkBasedDateTimeZone(String zoneId)
```
Comment (electrum) [inline src/main/java/io/airlift/jodabridge/JdkBasedDateTimeZone.java:29]: `private`

File: src/main/java/io/airlift/jodabridge/JdkBasedZoneInfoProvider.java [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    public static void registerAsJodaZoneInfoProvider()
+    {
+        // An alternative way of registering this Provider is by calling `DateTimeZone.setProvider`.
+        // However, that way, it won't be possible to tell whether any one has used DateTimeZone before this method is invoked.
+        // Setting it through the system property allows one to validate that.
+        System.setProperty("org.joda.time.DateTimeZone.Provider", JdkBasedZoneInfoProvider.class.getName());
+        if (!(DateTimeZone.getProvider() instanceof JdkBasedZoneInfoProvider)) {
+            throw new RuntimeException("This method must be invoked before any use of Joda");
```
Comment (electrum) [inline src/main/java/io/airlift/jodabridge/JdkBasedZoneInfoProvider.java]: Joda-Time

File: src/test/java/io/airlift/jodabridge/TestJdkBasedZoneInfoProvider.java [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+public class TestJdkBasedZoneInfoProvider
+{
+    @BeforeClass
+    protected void validateJodaZoneInfoProvider()
+    {
+        try {
+            JdkBasedZoneInfoProvider.registerAsJodaZoneInfoProvider();
+        }
+        catch (RuntimeException e) {
+            throw new RuntimeException("Set the following system property to JVM running the test: -Dorg.joda.time.DateTimeZone.Provider=com.facebook.presto.tz.JdkBasedZoneInfoProvider");
```
Comment (electrum) [inline src/test/java/io/airlift/jodabridge/TestJdkBasedZoneInfoProvider.java]: to -> for

File: src/main/java/io/airlift/jodabridge/JdkBasedDateTimeZone.java (hunk 13 above)
Comment (haozhun) [reply src/main/java/io/airlift/jodabridge/JdkBasedDateTimeZone.java:29]: The visibility needs to be at least package-private.

Comment (electrum) [reply src/main/java/io/airlift/jodabridge/JdkBasedDateTimeZone.java:29]: Ah, I misread the code and thought it was constructed here.

**Pull request merged**